### PR TITLE
Add support for Indonesia Kartu Multi Trip 

### PR DESCRIPTION
**Pull request by sybond**
Adding new support to read Indonesia CommuterLine's KMT cards which is using FeliCa. Basically just refactor `farebot-transit-edy`:

- Introduce `farebot-transit-kmt`
- Change service code for KMT
- Add checking to `getService` result to avoid `nullPointerException`
- Stripped `Trips` implementation, need to do some research to read historical data

### CODE REVIEW

File: farebot-app/build.gradle [hunk 1]
```diff
@@ -77,6 +77,7 @@ dependencies {
 
     implementation libs.dagger
     kapt libs.daggerCompiler
+    compile project(path: ':farebot-transit-kmt')
```
Comment (codebutler) [inline farebot-app/build.gradle]: move this with the others and use `implementation`

File: farebot-transit-kmt/.gitignore [hunk 2]
```diff
@@ -0,0 +1 @@
+/build
```
Comment (codebutler) [inline farebot-transit-kmt/.gitignore]: delete file

File: farebot-transit-kmt/proguard-rules.pro [hunk 3]
```diff
@@ -0,0 +1,21 @@
+# Add project specific ProGuard rules here.
```
Comment (codebutler) [inline farebot-transit-kmt/proguard-rules.pro]: delete file

File: farebot-transit-kmt/src/main/java/com/codebutler/farebot/transit/kmt/KMTTransitFactory.java [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+import com.codebutler.farebot.transit.Trip;
+import net.kazzz.felica.lib.Util;
+import java.util.ArrayList;
+import java.util.List;
+
+public class KMTTransitFactory implements TransitFactory<FelicaCard, KMTTransitInfo> {
+
+    //Taken from NXP TagInfo reader data
+
+    //This should be in the FeliCaLib from Klazz
```
Comment (codebutler) [inline farebot-transit-kmt/src/main/java/com/codebutler/farebot/transit/kmt/KMTTransitFactory.java]: is there a newer version of this library that includes this?

File: farebot-transit-kmt/src/main/java/com/codebutler/farebot/transit/kmt/KMTTransitInfo.java [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+    @Nullable
+    @Override
+    public List<Subscription> getSubscriptions() {
+        return null;
+    }
+
+    @NonNull
+    @Override
+    public String getCardName(@NonNull Resources resources) {
+        return "Kartu Multi Trip";
```
Comment (codebutler) [inline farebot-transit-kmt/src/main/java/com/codebutler/farebot/transit/kmt/KMTTransitInfo.java]: string resource

File: farebot-transit-kmt/src/main/java/com/codebutler/farebot/transit/kmt/KMTTrip.java [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+        return "KCI";
+    }
+
+    @Override
+    public String getAgencyName(@NonNull Resources resources) {
+        switch (getProcessType()) {
+            case 1:
+                return "Debit / Gate";
+            default:
+                return "Credit / Top Up";
```
Comment (codebutler) [inline farebot-transit-kmt/src/main/java/com/codebutler/farebot/transit/kmt/KMTTrip.java]: string resources

File: farebot-transit-kmt/src/main/java/com/codebutler/farebot/transit/kmt/KMTTrip.java [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+    }
+
+    @Override
+    public boolean hasTime() {
+        return getTimestampData() != null;
+    }
+
+    @Override
+    public String getRouteName(@NonNull Resources resources) {
+        return "JABODETABEK";
```
Comment (codebutler) [inline farebot-transit-kmt/src/main/java/com/codebutler/farebot/transit/kmt/KMTTrip.java]: should this be localized too?

File: farebot-transit-kmt/src/main/res/values/strings.xml [hunk 8]
```diff
@@ -0,0 +1,5 @@
+<resources>
+    <string name="app_name">KMT</string>
```
Comment (codebutler) [inline farebot-transit-kmt/src/main/res/values/strings.xml]: unused resource

File: settings.gradle [hunk 9]
```diff
@@ -1,7 +1,7 @@
 include 'nfc-felica-lib'
 project(':nfc-felica-lib').projectDir = new File('third_party/nfc-felica-lib')
 
-include ':farebot-base'
+include ':farebot-base', ':farebot-transit-kmt'
```
Comment (codebutler) [inline settings.gradle]: please keep the formatting consistent

File: farebot-transit-kmt/src/main/java/com/codebutler/farebot/transit/kmt/KMTTransitFactory.java (hunk 4 above)
Comment (sybond) [reply farebot-transit-kmt/src/main/java/com/codebutler/farebot/transit/kmt/KMTTransitFactory.java]: No. I've checked on latest update.

**Comment by codebutler** [comment]
done, thanks again!

**Pull request merged**